packages/ember-headless-form/src/components/-private/field.ts

```diff
@@ -123,7 +123,8 @@ export default class HeadlessFormFieldComponent<
   }
 
   get value(): DATA[KEY] {
-    return this.args.data[this.args.name];
+    // when @mutableData is set, data is something we don't control, i.e. might require old-school get() to be on the safe side
```

Comment:
I don't really know the history of the old school ember `get` versus the new getters...is there an article on it?

Reply:
Oh, you are blessed then! 😅

I don't know of an article that specifically focuses on that. But `get`/`set` (or `this.get()` and `this.set()`) were for a long part essential pieces of Ember before the Octane edition. They were required to access [Computed Properties (CP)](https://api.emberjs.com/ember/release/classes/ComputedProperty), which are still a thing, not yet deprecated, and probably still all around the monorepo codebase, but kinda out-of-favor and superseded by the Octane reactivity model of `@tracked` and native getters. A while ago the need for `get()` to access them was dropped, except for some special cases (in ember-data, idk if that is still required now, I hope not), but `set()` is still needed to set a CP. Also `get()` enabled nested property access, as in `get(myObj, 'some.nested.property')`.